test/setup-signin.js

```diff
@@ -0,0 +1,34 @@
+// This file sets up all subsequent tests.
+// It should be called first, as specified in `mocha.opts`
+// Most importantly, this file sets up global `page` and `browser` variables
+// for use across all test files under `./tests`.
+// It also signs into CRM based on credentials provided in `.env`. 
+const Puppeteer = require('puppeteer');
+require('dotenv').config();
+
+browser = null;
+page = null;
+
+// Launch puppeteer and instantiate browser and page objects.
+// Walk through CRM sign in pages until CRM landing page is visible.
+setTimeout(async function() {
```

Comment:
Why do we need setTimeout?

Reply:
There's a lot of waiting involved while signing in to the CRM, and sometimes it takes a long while. The setTimeout is for doing asynchronous operations, signin in this case, before any of the tests are run. 
https://mochajs.org/#delayed-root-suite

Reply:
@godfreyyeung gotcha, okay — I think [waitForSelector](https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#pagewaitforselectorselector-options) would be a little more resilient here. I'm mostly just trying to understand why it needs timing out for 0 seconds